Source/UrlRequest_Base.h

```diff
@@ -21,6 +21,20 @@ namespace UrlLib
             m_cancellationSource.cancel();
         }
 
+        void SetRequestBody(std::string requestBody) {
+            m_requestBody = requestBody;
+        }
+
+        void SetRequestHeader(std::string name, std::string value)
+        {
+            m_requestHeaders[name] = value;
+        }
+
+        std::unordered_map<std::string, std::string> GetAllResponseHeaders() const
```

Comment:
I would assume the return value is intended to be read-only? Would this be better to avoid copying the whole map?
```suggestion
        const std::unordered_map<std::string, std::string>& GetAllResponseHeaders() const
```

Reply:
Hmm, I noticed I also copy for the GetResponseHeader function. That ideally shouldn't be the case either. Nothing you need to do for GetResponseHeader, just FYI.